modules/cluster-management/src/main/java/org/apache/ignite/internal/cluster/management/ClusterManagementGroupManager.java

```diff
@@ -331,7 +337,8 @@ private void handleInit(CmgInitMessage msg, String senderConsistentId, long corr
     }
 
     private CompletableFuture<CmgRaftService> doInit(CmgRaftService service, CmgInitMessage msg) {
-        return service.initClusterState(createClusterState(msg))
+        return CompletableFuture.runAsync(() -> validateConfiguration(msg.clusterConfigurationToApply()))
```

Comment:
Again, this looks like an abuse, what's the point of dispatching this call on a default executor, when its instead should be executed in the current thread?